Controller/Payment/Index.php

```diff
@@ -29,20 +32,18 @@ public function execute()
                 "callback_url" => $this->_url->getUrl('satispay/callback/', [
                     "_query" => "payment_id={uuid}"
                 ]),
+                "redirect_url" => $this->_url->getUrl('satispay/redirect/'),
                 "metadata" => [
                     "order_id" => $order->getId(),
-                    "redirect_url" => $this->_url->getUrl('satispay/redirect/', [
-                        "_query" => "payment_id={uuid}"
-                    ])
                 ]
             ]);
-
-            $satispayUrl = 'https://online.satispay.com';
-            if (\SatispayGBusiness\Api::getSandbox()) {
-                $satispayUrl = 'https://staging.online.satispay.com';
+            $payment = $order->getPayment();
+            if (isset($payment)) {
+                // Set last transition id as the satispay payment id
+                $payment->setLastTransId($satispayPayment->id);
+                $this->orderRepository->save($order);
             }
```

Comment:
loggerei il caso in cui il payment non è settato perché è una situazione di errore che non ci aspettiamo succeda.